cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/TaskWorkerHttpHandlerInternal.java

```diff
@@ -74,21 +97,76 @@
 
   private final RunnableTaskLauncher runnableTaskLauncher;
   private final BiConsumer<Boolean, String> stopper;
+
   private final AtomicInteger inflightRequests = new AtomicInteger(0);
```

Comment:
Does this need to be atomic? All reference to it is in sync'ed block, right?

Reply:
The reason this need to be atomic is for the in run method.
```
    if (inflightRequests.incrementAndGet() > 1) {
      responder.sendStatus(HttpResponseStatus.TOO_MANY_REQUESTS);
      return;
    }
 ```